app/javascript/admin/adminModal.js

```diff
@@ -4,28 +4,32 @@
  * @function adminModal
  * @param {string} title The title of the modal.
  * @param {string} body The modal's content. May use HTML tags for styling.
- * @param {string} confirmBtnText The text for the modal's "Confirm" button.
- * @param {string} confirmBtnAction The function that fires when "Confirm" button is clicked.
- * @param {string} cancelBtnText The text for the modal's "Cancel" button.
- * @param {string} cancelBtnAction The function that fires when "Cancel" button is clicked.
+ * @param {string} leftBtnText The text for the modal's left button.
+ * @param {string} leftBtnAction The function that fires when left button is clicked.
+ * @param {string} rightBtnText The text for the modal's right button.
+ * @param {string} rightBtnAction The function that fires when right button is clicked.
+ * @param {string} leftBtnClasses Classes applied to left button.
+ * @param {string} rightBtnClasses Classes applied to right button.
  * @param {string} customAttr A custom data attribute name. Will be apprended to the "data-" part.
  * @param {string} customAttrValue The value of the custom attribute "customAttr".
  */
 const adminModal = (
   title,
   body,
-  confirmBtnText,
-  confirmBtnAction,
-  cancelBtnText,
-  cancelBtnAction,
+  leftBtnText,
```

Comment:
refactor (blocking): Typically you want 1-3 parameters in a function. More than that and it gets hairy. Having one parameter with properties would work better here, like the props property in React/Preact. With that you get named properties as well which means the order they're defined in won't matter.

```javascript
const adminModal = ({
  title,
  body,
  leftBtnText,
  leftBtnAction,
  rightBtnText,
  rightBtnAction,
  leftBtnClasses,
  rightBtnClasses,
  customAttr = null,
  customAttrValue = null,
}) => `
...
```

nitpick (non-blocking): Prefer a [function expression](https://developer.mozilla.org/en-US/docs/web/JavaScript/Reference/Operators/function) over an [arrow function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) unless you need access to the lexical scope. i.e. access to the `this` property.

Reply:
Great point, @nickytonline! Will do!